packages/base/src/widget.ts

```diff
@@ -88,6 +88,21 @@ export class WidgetModel extends Backbone.Model {
     };
   }
 
+  /**
+   * Generate extra mime bundle for this widget
+   */
+  async generateMimeBundle() {
+    return {};
+  }
+
+  /**
+   * Whether the `generateMimeBundle` output should
+   * overwrite the mimeBundle or extend it
+   */
+  shouldOverwriteMimeBundle(): boolean {
+    return false;
+  }
+
```

Comment:
What about generateMimeBundle making the default mimebundle (with the widget in it).
subclases can then do:
```javascript
return {'text/html': view.el.outerHTM, ... await super.generateMimeBundle()}
// or
return {'text/html': view.el.outerHTM}
```
depending on if they want to overwrite it. Does that make sense?

Reply:
Makes sense!